src/Serialization.SystemTextJson/GraphQLRequestDeserializer.cs

```diff
@@ -0,0 +1,89 @@
+using GraphQL.Server.Common;
+using System;
+using System.Buffers;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace GraphQL.Server.Serialization.SystemTextJson
+{
+    public class GraphQLRequestDeserializer : IGraphQLRequestDeserializer
+    {
+        private readonly JsonSerializerOptions _serializerOptions = new JsonSerializerOptions();
+
+        public GraphQLRequestDeserializer(Action<JsonSerializerOptions> configure)
+        {
+            configure(_serializerOptions);
+        }
+
+        public async Task<GraphQLRequestDeserializationResult> FromBodyAsync(Stream stream, long? contentLength)
+        {
+            int length;
+            byte[] jsonBytes;
+            var sharedArrayPool = ArrayPool<byte>.Shared;
+            bool wasRented = false;
+            if (contentLength.HasValue)
+            {
+                length = (int)contentLength;
+                jsonBytes = sharedArrayPool.Rent(length);
+                wasRented = true;
+                await stream.ReadAsync(jsonBytes, 0, length);
```

Comment:
I'm now renting out an array pool if I have a ContentLength given, then async reading the request body into it.

Reply:
One concern I have here is that I can only rent out with an `int`, but the contentLength is a `long`. Not sure how to handle that though. Update the `if` to check it's less than `int.MaxValue`? Can't imagine it ever would be, but you never know

Reply:
If this becomes a problem, then leave it to future generations of developers.

Reply:
I propose to check contentLength to int.MaxValue and throw NotSupportedException. It will be interesting to see real scenarios for using this package when a request of 2 GB in size is transmitted as a payload :)